packages/js/data/src/product-variations/actions.ts

```diff
@@ -46,6 +47,48 @@ export function generateProductVariationsSuccess( key: IdType ) {
 	};
 }
 
+type ConfigEntity = {
+	name: string;
+	kind: string;
+	baseURL: string;
+	baseURLParams: {
+		context: string;
+	};
+};
+
+/**
+ * Refreshes the data within the core entity data store of products without invalidating.
+ *
+ * @param {string|number} productId productId of product to refresh.
+ */
+export function* refreshProductVariationData( productId: string | number ) {
+	const configs: ConfigEntity[] = yield controls.select(
+		'core',
+		'getEntitiesConfig',
+		'postType'
+	);
+	const entityConfig = configs.find(
+		( config ) => config.name === 'product' && config.kind === 'postType'
+	);
+	if ( ! entityConfig ) {
+		return;
+	}
+	const path = addQueryArgs( entityConfig.baseURL + '/' + productId, {
+		...entityConfig.baseURLParams,
+	} );
+	const response: Product = yield apiFetch( {
+		path,
+		method: 'GET',
+	} );
+	yield controls.dispatch(
+		'core',
+		'receiveEntityRecords',
+		'postType',
+		'product',
+		[ response ]
+	);
+}
```

Comment:
We use `apiFetch` here to fetch the new product and triggering the `receiveEntityRecords` action so the store gets updated correctly with the latest product data.